javarosa-dag.rst

```diff
@@ -0,0 +1,304 @@
+
+If such triggerable already exists, ``IDag`` does not add a new instance but updates the existing``Triggerable`` one's context so it points to the 
+highest common root. For the above example, the highest common root for ``nodeA`` and ``nodeB`` calculations is ``/data``.
+The passed instance ``t`` is ignored and a previously added ``Triggerable`` one is returned and should be used for the further processing.
+The context is used later when constructing a DAG structure.
+
+If such triggerable does not exist, it is added to the ``unorderedTriggerables`` list and the ``triggerIndex`` map.
+The passed triggerable ``t`` instance is returned.
+The list is later used to construct the topoligically sorted list ``triggerablesDAG``.
+The ``triggerIndex`` maps the triggers to the triggerables.
+
+Each ``Triggerable`` instance comes with a list of targets which should be updated with the evaluation result.
+In the above example, ``pow(/data/nodeC, 2)`` calculation have two targets: ``nodeA`` and ``nodeB``.
+When a ``Triggerable`` instance is constructed, the targets list cannot be populated because each ``bind`` node
+is parsed separately and other targets - if any - for the same triggerable are unknown. Moreover,
+the newly constructed ``Triggerable`` may be ignored if another instance that represents the same triggerable
+have been already added to the DAG.
+
+.. Actually, the above is not entirely true. I think it can be refactored so the targets are populated during 
+.. the bind processing. But a particular attention must be paid to call ``IDag.addTriggerable(Triggerable t)`` before
+.. ``Triggerable.addTarget(TreeReference target)`` because  ``IDag.addTriggerable(Triggerable t)` may return a previously added
+.. instance and there fore ``t`` is ignored.
+
+The ``targets`` list population is postponed and takes place during the form instance parsing, 
+in the ``FormInstanceParser.applyInstanceProperties(FormInstance instance)`` method.
+
+Finalization and constructing the DAG
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once all ``bind`` nodes are parsed and all ``Triggerable`` collected, the ``IDag`` can be finalized and
+the actual DAG structure can be constructed. The ``finalizeTriggerables(FormInstance mainInstance, EvaluationContext evalContext)``
+method is responsible for the finalization. Once it completes, ``triggerablesDAG`` and ``conditionRepeatTargetIndex`` becomes
+populated and valid. 
+
+* ``triggerablesDAG`` stands for the topologically ordered list of ``Triggerable`` instances. All evaluation takes place according to the order in this list.
+* ``conditionRepeatTargetIndex`` maps a repeat group reference (``TreeReference``) to its``Condition`` that determines the relevance.  
+
+
+Triggering
+----------
+
+Evaluation of triggerables is triggered when:
+
+* A form is loaded
+* A value of a node changes (an answer is committed)
+* A repeat group is added
+* A repeat group is deleted
```

Comment:
A repeat group is added or deleted

What do you think? I personally have a mild preference for the refactored form.